core/src/main/scala/kafka/log/Log.scala

```diff
@@ -75,7 +75,7 @@ case class LogAppendInfo(var firstOffset: Long,
  *
  */
 @threadsafe
-class Log(val dir: File,
+class Log(@volatile var dir: File,
```

Comment:
Can you verify that this does not stronger protection? E.g., I searched for usages and log roll for instance accesses it. It may be harmless but can you confirm?